### ISSUE TITLE
Support for Mapsforge's .poi format

### ISSUE THREAD

**Issue by hshagsfgd** [tumic0/GPXSee#420]
**TL;DR**

Support for Mapsforge's `.poi` format.

---
After [learning](https://github.com/tumic0/GPXSee/issues/417#issuecomment-1011880510) that Mapsforge's  `.map` format is supported, I have opened several such files from different [providers](https://github.com/mapsforge/mapsforge/blob/master/docs/Mapsforge-Maps.md) and compared them, and unfortunately found out, that while map 1 would have certain POIs/names, map 2 would have different POIs/names displayed (not sure if they are actual baked in POIs or just part of some kind of names, like street names, since I didn't open a POI file, just the `.map` file) (zooming around would still not display certain POIs, they may not be in the `.map` file at all). Fortunately some of the providers also offer/include a `.poi` file, so after I opened the `.map` file, I also tried to open the `.poi` file from the `POI` menu to add the missing POIs, but unfortunately the format seems not to be supported and I get `Unknown format` error.

**Comment by tumic0** [comment]
Mapsforge POI files are unusable in GPXSee as they (usually) contain way too much data that can not be processed in the same way like all other POI files. For example for my country, the reletively small Czech Republic, the POI file has ~390k entries. And that's not gonna "move" on any todays computer (on an average notebook, you get stuck at ~20k points). And there are much bigger (in terms of POIs number) countries like Germany... The POI file, despite its name, is in fact not a POI file, but a dump of all OSM point objects.

To do something with the files, you would have to completely change the logic of POI displaying - load/show them only on some high map zoom levels or to filter them somehow. But that is out of scope of GPXSee and if you do that than you can most probably save the result in another format that GPXSee understands.

**Comment by hshagsfgd** [comment]
No one is going to want to see all POIs at once and that's why Mapsforge's `.poi` format exist. It's a database format/file and makes search queries basically instant (even on older smartphones), for each zoom level/area/depth. Yes, it's not as simple as a simple `.gpx` XML file, but it's more flexible.

>The POI file, despite its name, is in fact not a POI file, but a dump of all OSM point objects

And some point objects can be of interest ;)

I feel like you closed this one too early (even if the program's name may imply only GPX support, but it already supports not only that one), even if it's more work to implement. Maybe someone reads this and wants to contribute patches to add support for it.

PS: I found [this](https://github.com/simao/osm-gpx) can convert `.pbf` to `.gpx`. And if one wants to limit the number of POIs, here's an [extractor](https://extract.bbbike.org). Although the converter didn't work for me so I'm opening a new issue.

**Comment by tumic0** [comment]
Even when used as POI files in GPXSee, the Mapsforge POI files are unusable in GPXSee, see the attached example. Some filtering is necessary to work reasonable with the files and that is out of scope of GPXSee. But like usual, feel free to implement this feature in your own fork of GPXSee.

![POI](https://user-images.githubusercontent.com/11480505/150222188-8f4bd63e-70a0-4e48-baf2-feb99b033f9d.png)
